StripeIdentity/StripeIdentity/Source/NativeComponents/Detectors/LaplacianBlurDetector.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  LaplacianBlurDetector.swift
+//  StripeIdentity
+//
+//  Created by Chen Cen on 8/22/23.
+//
+
+import Accelerate
+import Foundation
+import Metal
+import MetalKit
+import MetalPerformanceShaders
+@_spi(STP) import StripeCameraCore
+import Vision
+
+/// Detector to determine if an image is blurry based on variance of the  laplacian method implemented with Metal.
+final class LaplacianBlurDetector {
+    struct Output: Equatable {
+        let isBlurry: Bool
+        let variance: Float
+    }
+
+    let blurThreshold: Float
+
+    private let mtlDevice = MTLCreateSystemDefaultDevice()
+    lazy var mtlCommandQueue = {
+        return mtlDevice?.makeCommandQueue()
+    }()
+
+    init(blurThreshold: Float) {
+        self.blurThreshold = blurThreshold
+    }
+
+    func calculateBlurOutput(inputImage: CGImage) -> Output {
+        guard let mtlCommandQueue = self.mtlCommandQueue, let mtlDevice = self.mtlDevice
+        else{
+            return Output(isBlurry: false, variance: 0)
+        }
+        // Create a command buffer for the transformation pipeline
+        let commandBuffer = mtlCommandQueue.makeCommandBuffer()!
```

Comment:
Worth double-checking whether force cast "!" is the correct choice:
1. Is there a 100% certainty that this will never fail?
2. If (1) is uncertain, is the preferred way that a failure is handled is an app crash? 

If not, maybe there's better ways to fail, but I leave that up to the you!

Reply:
Added guard check and default value to return